libpod/kube.go

```diff
@@ -1134,16 +1134,19 @@ func determineCapAddDropFromCapabilities(defaultCaps, containerCaps []string) *v
 }
 
 func (c *Container) capAddDrop(caps *specs.LinuxCapabilities) *v1.Capabilities {
-	// Combine all the container's capabilities into a slice
-	containerCaps := make([]string, 0, len(caps.Ambient)+len(caps.Bounding)+len(caps.Effective)+len(caps.Inheritable)+len(caps.Permitted))
-	containerCaps = append(containerCaps, caps.Ambient...)
-	containerCaps = append(containerCaps, caps.Bounding...)
-	containerCaps = append(containerCaps, caps.Effective...)
-	containerCaps = append(containerCaps, caps.Inheritable...)
-	containerCaps = append(containerCaps, caps.Permitted...)
-
-	calculatedCaps := determineCapAddDropFromCapabilities(c.runtime.config.Containers.DefaultCapabilities, containerCaps)
-	return calculatedCaps
+	if caps != nil {
```

Comment:
Nit: this would be better if it was

if caps == nil {
      return nil
}
...

Reply:
Good idea - I also added a comment to explain why its there.